.ci/README.md

```diff
@@ -32,80 +32,128 @@ It's possible that a consumer of the e2e tests would need to define a specific l
 
 The following variables need to be exported:
 
-- *RUN_ID*: This is a unique identifying ID for the current run. It can be an arbitrary name or something like this:
-
-```
-export RUN_ID=$(uuidgen|cut -d'-' -f1)
-```
-
 - *AWS_SECRET_ACCESS_KEY*: AWS secret access key
 - *AWS_ACCESS_KEY_ID*: AWS access key id
 
 Install python deps:
 
-```
-> python3 -mvenv venv
-> venv/bin/pip3 install ansible requests boto3 boto
-> venv/bin/ansible-galaxy install -r .ci/ansible/requirements.yml
+```shell
+make -C .ci setup-env
 ```
 
-### Deploy stack
+It will create a `.runID` under the `.ci` directory. It will contain an unique identifier for your machines, which will be added as a VM tag.
 
+We are able to run the Elastic Stack and all the test suites in AWS instances, so whenever a build failed we will be able to recreate the same machine and access it to inspect its state: logs, files, containers... For that, we are enabling SSH access to those ephemeral machines, which can be created following this guide.
+
+But you must first understand that there are two types of Cloud machines: 
+1) the VM running the stack, and 
+2) the VMs where the Elastic Agent will be installed and enrolled into the stack.
+
+### Create and configure the stack VM
+
+This specialised VM starts Elasticsearch, Kibana and Fleet Server using Docker Compose, but instead of invoking the compose file directly, it uses the test framework to do it. Why? Because we need to wait for Elasticsearch to be ready and request an API Token to be passed to the Fleet Server container. And [we do this with code](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/e2e/_suites/fleet/fleet.go#L631-L748).
+
+The VM is a Debian AMD64 machine, as described [here](https://github.com/elastic/e2e-testing/blob/4517dfa134844f720139d6bab3955cc8d9c6685c/.ci/.e2e-platforms.yaml#L3-L7).
+
+```shell
+export SSH_KEY="PATH_TO_YOUR_SSH_KEY_WITH_ACCESS_TO_AWS"
+make -C .ci create-stack
 ```
-> venv/bin/ansible-playbook .ci/ansible/playbook.yml \
-    --private-key="$HOME/.ssh/id_rsa" \
-    --extra-vars "nodeLabel=stack nodeImage=ami-0d90bed76900e679a nodeInstanceType=c5.4xlarge" \
-    --extra-vars "runId=$RUN_ID workspace=$HOME/Projects/e2e-testing/ sshPublicKey=$HOME/.ssh/id_rsa.pub" \
-    --ssh-common-args='-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' \
-    -t provision-stack
-```
 
-Make note of the IP address displayed in the ansible summary.
+A `.stack-host-ip` file will be created in the `.ci` directory of the project including the IP address of the stack instance. Check it out from that file, or make a note of the IP address displayed in the ansible summary, as you'll probably need it to connect your browser to open Kibana, or to SSH into it for troubleshooting.
```

Comment:
I know it's a bit lazy, but we could have a `make show-stack-ip` that `cat` this file. What do you think?

Reply:
Done

Reply:
```shell
$ make -C .ci show-stack
Stack IP address    : 18.220.137.250
Stack type          : debian_amd64
Stack instance type : c5.4xlarge
Stack SSH user      : admin
```